control/tests/test_deliver.py

```diff
@@ -0,0 +1,169 @@
+"""
+Veda Delivery unit tests
+"""
+from mock import PropertyMock, patch
+import os
+import responses
+import unittest
+
+from django.test import TestCase
+
+from control.veda_deliver import VedaDelivery
+from control.veda_val import VALAPICall
+from control.veda_file_ingest import VideoProto
+from VEDA_OS01.models import Course, Video, Destination, Encode, URL
+from VEDA.utils import get_config
+
+CONFIG_DATA = get_config('test_config.yaml')
+
+
+class VedaDeliverRunTest(TestCase):
+    """
+    Deliver Run Tests
+    """
+    def setUp(self):
+        self.veda_id = 'XXXXXXXX2014-V00TES1'
+        self.encode_profile = 'hls'
+        self.course = Course.objects.create(
+            institution='XXX',
+            edx_classid='XXXXX',
+            course_name=u'Intro to VEDA',
+            local_storedir=u'This/Is/A/testID'
+        )
+        self.video = Video.objects.create(
+            inst_class=self.course,
+            edx_id=self.veda_id,
+            client_title='Test Video',
+            video_orig_duration='00:00:10.09',  # This is known
+            pk=1
+        )
+        self.destination = Destination.objects.create(
+            destination_name='TEST'
+        )
+        self.encode = Encode.objects.create(
+            encode_destination=self.destination,
+            profile_active=True,
+            encode_suffix='HLS',
+            product_spec=self.encode_profile,
+            encode_filetype='HLS'
+        )
+        self.DeliverInstance = VedaDelivery(
+            veda_id=self.veda_id,
+            encode_profile=self.encode_profile,
+            CONFIG_DATA=CONFIG_DATA
+        )
+        # VAL Patching
+        responses.add(responses.POST, CONFIG_DATA['val_token_url'], '{"access_token": "1234567890"}', status=200)
+        responses.add(
+            responses.GET,
+            CONFIG_DATA['val_api_url'] + '/XXXXXXXX2014-V00TES1',
+            status=200,
+            json={'error': 'null', 'courses': [], 'encoded_videos': []}
+        )
+        responses.add(responses.PUT, CONFIG_DATA['val_api_url'] + '/XXXXXXXX2014-V00TES1', status=200)
+        self.VP = VideoProto(
+            client_title='Test Video',
+            veda_id=self.veda_id
+        )
+        with patch.object(VALAPICall, '_AUTH', PropertyMock(return_value=lambda: CONFIG_DATA)):
+            self.VAC = VALAPICall(
+                video_proto=self.VP,
+                val_status='complete'
+            )
+
+    @patch('control.veda_val.VALAPICall._AUTH', PropertyMock(return_value=lambda: CONFIG_DATA))
+    @responses.activate
```

Comment:
we don't need to activate `responses` for this test case.

Reply:
@Qubad786 It's failing without it -- it makes the call deeper in the run function, which is a little bit redundant from the `test_update_data` function

Reply:
ah, got it, please ignore this one.